src/crypto/crypto_context.cc

```diff
@@ -190,18 +197,93 @@ int SSL_CTX_use_certificate_chain(SSL_CTX* ctx,
 
 }  // namespace
 
+void ReadSystemStoreCertificates(
+    std::vector<std::string>* system_root_certificates) {
+#ifdef _WIN32
+  const HCERTSTORE hStore = CertOpenSystemStoreW(0, L"ROOT");
```

Comment:
Worth taking a look at this change to openjdk and thinking about is the system store enough:
https://github.com/openjdk/jdk/commit/5e5500cbd79b40a32c20547ea0cdb81ef6904a3d

https://bugs.openjdk.org/browse/JDK-6782021

There's also the Jetbrains implementation that accesses a number of stores and aggregates them:
https://github.com/JetBrains/jvm-native-trusted-roots/blob/trunk/src/main/java/org/jetbrains/nativecerts/win32/Crypt32ExtUtil.java#L30-L36

I've had confirmation that the jetbrains one doesn't work with intermediate certificates, awaiting confirmation with the OpenJDK one.